keras_nlp/samplers/sampler.py

```diff
@@ -358,20 +385,26 @@ def one_step(
                 return current_index, prompt, mask
             return [current_index, prompt, mask, cache]
 
+        def loop_cond(current_index, prompt, mask, cache=None):
+            end_token_seen = (prompt == end_token_id) & (
```

Comment:
Don't we need to handle end_token_id == None?

Reply:
good call, fixed.